client/ayon_houdini/plugins/publish/validate_render_product_paths_unique.py

```diff
@@ -0,0 +1,109 @@
+import inspect
+from collections import defaultdict
+from typing import List
+
+import pyblish.api
+import clique
+import hou
+
+from ayon_core.pipeline import (
+    OptionalPyblishPluginMixin,
+    PublishValidationError
+)
+
+from ayon_houdini.api import plugin
+from ayon_houdini.api.action import SelectInvalidAction
+
+
+class ValidateRenderProductPathsUnique(plugin.HoudiniContextPlugin,
+                                       OptionalPyblishPluginMixin):
+    """Validate that render product paths are unique.
+
+    This allows to catch before rendering whether multiple render ROPs would
+    end up writing to the same filepaths. This can be a problem when rendering
+    because each render job would overwrite the files of the other at
+    rendertime.
+
+    """
+    order = pyblish.api.ValidatorOrder
+    families = ["usdrender"]
+    hosts = ["houdini"]
+    label = "Unique Render Product Paths"
+    actions = [SelectInvalidAction]
+    optional = True
+
+    def process(self, context):
+        if not self.is_active(context.data):
+            return
+
+        invalid = self.get_invalid(context)
+        if not invalid:
+            return
+
+        node_paths = [node.path() for node in invalid]
+        node_paths.sort()
+        invalid_list = "\n".join(f"- {path}" for path in node_paths)
+        raise PublishValidationError(
+            "Multiple instances render to the same path. "
+            "Please make sure each ROP renders to a unique output path:\n"
+            f"{invalid_list}",
+            title=self.label,
+            description=self.get_description()
+        )
+
+    @classmethod
+    def get_invalid(cls, context) -> "List[hou.Node]":
+        # Get instances matching this plugin families
+        instances = pyblish.api.instances_by_plugin(list(context), cls)
+        if len(instances) < 2:
+            return []
```

Comment:
so, it won't error if I have one instance with separate AOVs that don't have unique paths?
e.g. beauty, P and N that all have the same path. (most likely due to miss configured)

Reply:
Hmm - correct. But I'm also worried that we'd suddenly be taking too much control there otherwise. Plus the whole error report and messaging would need to differ too if it could suddenly even be within the one instance.

It's also a much less common edge case plus it'd behave exactly similar if user were to just render the single ROP without AYON.

I'll ignore this for now.

Reply:
Just for reference, maybe I'm overthinking. but when having two render rops 
first one with the case mentioned above and the second is properly configured, the validator will fail. 
and here's how the log will look like: 
```
Multiple instances output to path: //storage/work/ayon/RnD/features/deadline_submissions/work/Generic/pyblish/renders/mantra_ropTest/mantra_ropTest.%d.exr [1001-1005]

Multiple instances render to the same path. Please make sure each ROP renders to a unique output path:
- /out/mantra_ropTest1
```